src/reducers/app.js

```diff
@@ -7,18 +7,17 @@ import { combineReducers } from 'redux';
 import { tabSlugs } from '../app-logic/tabs-handling';
 
 import type { TabSlug } from '../app-logic/tabs-handling';
-import type { Action } from '../types/store';
 import type {
   AppState,
   AppViewState,
   IsSidebarOpenPerPanelState,
   Reducer,
 } from '../types/reducers';
 
-function view(
-  state: AppViewState = { phase: 'INITIALIZING' },
-  action: Action
-): AppViewState {
+const view: Reducer<AppViewState> = (
+  state = { phase: 'INITIALIZING' },
```

Comment:
I noticed that Flow doesn't check that the default parameter is correct if we don't type it explicitely. Maybe [this is a bug in Flow](https://flow.org/try/#0KYDwDg9gTgLgBDAnmYcDKEC2wBiBXAOwGMYBLCAgHgBUA+OAXjgApqBKR+zUkYAEwDcAKCIUAzvABmhIgC50WXDLIVKEqKQIBzek2ZgAhlAOZGcAMwcG9AN5C4cUQTEQANsAB0riFv1GTHppErnh8wGLMAESSEBCRbGzCAL4CQA), or maybe we don't understand how to type it. It looks like Flow disregards the type in this case.

(to check this in our code, I changed the `default` case, and the initial `if`, to return litteral objects). 

_But_ it is indirectly checked with `default: return state`. Is it good enough? Probably yes, because there's no point having a default state if we don't return it somewhere.

 I checked other cases that could be problematic because of this (for example: check that flow properly checks enum states like the PreviewSelection) and Flow actually checked them properly too. So I think we're good.

Reply:
Thanks for the extra scrutiny here.